app/pages/project/workflow-selection.jsx

```diff
@@ -69,10 +69,11 @@ class WorkflowSelection extends React.Component {
     } else if (project.configuration && project.configuration.default_workflow) {
       selectedWorkflowID = project.configuration.default_workflow;
     } else {
-      selectedWorkflowID = this.selectRandomWorkflow(project);
+      selectedWorkflowID = this.selectRandomActiveWorkflow(project);
     }
 
-    this.getWorkflow(selectedWorkflowID, activeFilter);
+    if (selectedWorkflowID) return this.getWorkflow(selectedWorkflowID, activeFilter);
```

Comment:
`this.getWorkflow` doesn't return a value, so I think this will return undefined. Is it intended to return a promise?

Reply:
Not exactly. I just wanted to make sure the following line's console message only happened if `selectedWorkflowID` is falsey. But good point, I'll make sure `getWorkflow` returns something.

Reply:
in #4670 I found that updating a method to return a promise made it easier to test, because I could look at how the promise resolved. Probably not relevant here because `getWorkflow` is stubbed in the tests, but it might come in useful.